tests/ui/if_let_mutex.rs

```diff
@@ -0,0 +1,17 @@
+#![warn(clippy::if_let_mutex)]
+
+use std::sync::Mutex;
+
+fn do_stuff<T>(_: T) {}
+
+fn if_let() {
+    let m = Mutex::new(1u8);
+    if let Err(locked) = m.lock() {
+        do_stuff(locked);
+    } else {
+        let lock = m.lock().unwrap();
```

Comment:
You should add a check to the `ArmVisitor`, that the mutexes are the same. You can do this with `utils::hir_utils::SpanlessEq`